ingraph-engine/ingraph-engine-ingraph-ire/src/main/scala/ingraph/ire/Indexer.scala

```diff
@@ -111,12 +111,13 @@ class Indexer {
     edgeLookup.remove(id)
   }
 
-  def vertexById(id: Long): IngraphVertex = vertexLookup(id)
-  def edgeById(id: Long): IngraphEdge = edgeLookup(id)
-  def verticesByLabel(label: String): Seq[IngraphVertex] = vertexLabelLookup(label)
-  def edgesByLabel(label: String): Seq[IngraphEdge] = edgeLabelLookup(label)
-  def getNumberOfVerticesWithLabel(label: String): Int =  vertexLabelLookup(label).size
-  def getNumberOfEdgesWithLabel(label: String): Int = edgeLabelLookup(label).size
+  def vertexById(id: Long): Option[IngraphVertex] = vertexLookup.get(id)
+  def edgeById(id: Long): Option[IngraphEdge] = edgeLookup.get(id)
+  def vertices(): Seq[IngraphVertex] = vertexLabelLookup.values.flatten.toSeq
```

Comment:
Not sure what you mean. Would this imply an immutable indexer? 
The defs are only evaluated when you call them.